### ISSUE TITLE
Add support for CRI-O container runtime

### ISSUE THREAD

**Issue by hakman** [kubernetes/kops#9712]
**1. Describe IN DETAIL the feature/behavior/change you would like to see.**

Alternative CRI-shim / runtime support was a popular request https://github.com/kubernetes/kops/issues/5524.
Support for containerd was added in Kops v1.18. Adding support for CRI-O is a bit harder compared to containerd, but pretty similar.

**2. Feel free to provide a design supporting your feature request.**

The main PR for adding contained support https://github.com/kubernetes/kops/pull/7986 can be used as an example.

**Comment by bharath-123** [comment]
@hakman is this being actively worked on right now? Would love to pick this up if not. Have already submitted a few PRs to kops. I am sure I will be challenged here.

**Comment by hakman** [comment]
> @hakman is this being actively worked on right now? Would love to pick this up if not. Have already submitted a few PRs to kops. I am sure I will be challenged here.

Why not, give it a try. If you see something out of place in the example PR, keep in mind that it was one of my first PRs here 😄 .

**Comment by bharath-123** [comment]
yes! Will try this out! hahaha will let you know if I find anything in the example PR :D

**Comment by BartoszZawadzki** [comment]
What is the status of this?

**Comment by bharath-123** [comment]
Hi @BartoszZawadzki . This feature is targeted for release in kOps 1.21. https://github.com/kubernetes/kops/pull/10662 have raised a WIP PR. we need to change protokube to a systemd process for this to work. So we targetted this work for kOps 1.21
